qiskit_machine_learning/utils/loss_functions/kernel_loss_functions.py

```diff
@@ -0,0 +1,143 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+""" Kernel Loss utilities """
+
+from functools import partial
+from abc import ABC, abstractmethod
+from typing import Sequence, Iterable, Callable
+
+import numpy as np
+from sklearn.svm import SVC
+
+# Prevent circular dependencies caused from type checking
+from ...kernels import QuantumKernel
+
+
+class KernelLoss(ABC):
+    """
+    Abstract base class for computing the loss of a kernel function.
+    Unlike many loss functions, which only take into account the labels and predictions
+    of a model, kernel loss functions may be a function of internal model parameters or
+    quantities that are generated during training.
+    """
+
+    def __call__(
+        self,
+        parameter_values: Sequence[float],
+        quantum_kernel: "QuantumKernel",
+        data: np.ndarray,
+        labels: np.ndarray,
+    ) -> float:
+        """
+        This method calls the ``evaluate`` method. This is a convenient method to compute loss.
+        """
+        return self.evaluate(parameter_values, quantum_kernel, data, labels)
+
+    def get_variational_callable(
+        self, quantum_kernel: QuantumKernel, data: Iterable, labels: Iterable
+    ) -> Callable[[Sequence[float]], float]:
+        """
+        Return a callable variational loss function given some inputs. The sole input to the
+        callable should be an array of numerical feature map parameter values, and the output should
+        be a numerical loss value.
+        """
+        raise NotImplementedError
```

Comment:
Don't really get why this one is required.

Reply:
Because other kernel loss functions may not need to take the same inputs in order to evaluate loss. Requiring this function be implemented, ensures that we can retrieve the loss by getting a callable that takes in only the set of parameter values and returns the loss. The inputs to this will always be the kernel, data, and labels, and any additional fields the loss needs would need to be held in the class